src/gas-api.js

```diff
@@ -4,215 +4,229 @@ class Gas {
+
+        if (context.projectId.length != 33) {
+          reject(new Error('cant not get project ID'));
+        }
+        const promises = ids.map((id) => {
+          return new Promise((resolve, reject) => {
+            const payload = `7|1|9|${getBaseUrl()}\|${context.gasToken}|_|getFileContent|l|${id}|${context.id}|${context.projectId}|m|1|2|3|4|1|5|5|6|7|8|9|0|0|`;
+            $.ajax({
+                url: context.gasUrl,
+                headers: context.gasHeaders,
+                method: 'POST',
+                crossDomain: true,
+                data: payload,
+                dataType: 'text'
+              })
+              .then((response) => {
+                if (!response.startsWith('//OK')) reject(new Error('get apps script code failed'));
+                //evil eval, but it's simple to get the object since it's not valid json object
+                const codeContent = eval(response.slice(4)).filter((e) => {
+                  return typeof (e) === 'object';
+                })[0];
+                resolve({
+                  file: codeContent[codeContent.length - 7],
+                  content: codeContent[codeContent.length - 10],
+                  id: id
+                });
+              })
+              .fail(reject);
+          })
+        });
+        return Promise.all(promises);
+      })
+      .then((responses) => {
+        context.fileIds = responses.reduce((hash, elem) => {
+          if (elem) hash[elem.file] = elem.id;
+          return hash;
+        }, {});
+        return responses;
+      })
+  }
 
-      if (context.projectId.length != 33) {
-        reject(new Error('cant not get project ID'));
-      }
-      const promises = ids.map((id) => {
-        return new Promise((resolve, reject) => {
-          const payload = `7|1|9|${getBaseUrl()}\|${context.gasToken}|_|getFileContent|l|${id}|${context.id}|${context.projectId}|m|1|2|3|4|1|5|5|6|7|8|9|0|0|`;
-          $.ajax({
+  gasCreateFile(file, type) {
+    const typeId = type === context.config.filetype ? 0 : 2;
+    const payload = `7|1|7|${getBaseUrl()}\|${context.gasToken}|_|makeNewFile|1a|i|${file}|1|2|3|4|2|5|6|7|6|${typeId}|`;
+    return new Promise((resolve, reject) => {
+        $.ajax({
             url: context.gasUrl,
             headers: context.gasHeaders,
             method: 'POST',
             crossDomain: true,
             data: payload,
             dataType: 'text'
           })
-          .then((response) => {
-            if (!response.startsWith('//OK'))  reject(new Error('get apps script code failed'));
-            //evil eval, but it's simple to get the object since it's not valid json object
-            const codeContent = eval(response.slice(4)).filter((e) => {
-              return typeof(e) === 'object';
-            })[0];
-            resolve({file : codeContent[codeContent.length - 7], content: codeContent[codeContent.length - 10], id : id });
+          .then(resolve)
+          .fail((err) => {
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/leonhartX/gas-github/blob/15f4157afeba0560f3f12968d583a6f4fceda1cf/src/gas-api.js#L161"></span>‘err’ is defined but never used.</td>
    </tr>
  </tbody>
</table>